### ISSUE TITLE
Many more allocations than encoding/json

### ISSUE THREAD

**Issue by arthurprs** [ugorji/go#134]
Running this [gist](https://gist.github.com/arthurprs/ed5ed9486850f01a0d7b) I get the following

```
BenchmarkJsonDecodeStd-8      	  100000	     19815 ns/op	     432 B/op	       8 allocs/op
BenchmarkJsonDecode-8         	  100000	     15148 ns/op	    1584 B/op	      50 allocs/op
BenchmarkJsonDecodeReader-8   	  100000	     24196 ns/op	    2914 B/op	      52 allocs/op
BenchmarkJsonDecodeReaderStd-8	   50000	     25241 ns/op	    4000 B/op	      13 allocs/op
```

Even if it's still faster, it's allocating much more. Am I missing something?

**Comment by ugorji** [comment]
It's not a bug that it allocates more.

You could help by doing some analysis and putting up the information. For now, there's different reasons why it could allocate more e.g. we use a few pointer structures when you create a Decoder, etc.

Also, it's hard to parse the gist you have. The use of sync.Pool and so on makes it unclear exactly what I'm looking at.

Please try to do some more analysis and ping me back on this, and I can look at it. For now, closing as a question, and NeedMoreInfo.
